### DISCUSSION TITLE
Plesk panel vds server socket.io is not working

### DISCUSSION THREAD

**Discussion by 0xBeycan** [socketio/socket.io#3772] [Q&A]
I developed a node js project in local, everything was working fine, but I could not install it on the server. More precisely, I installed it, but socket.io doesn't connect anyway.

```
// Create server
const server = require('http').createServer();

const io = require('socket.io')(server,{
    cors: {
        origin: "http://213.238.181.167",
        methods: ["GET", "POST"]
    }
});

io.on('connection', (socket) => {
    
    // First connection
    console.log("socket.io connected"); // Not working
 
});
  
server.listen(3000, () => {
    console.log("listen port: 3000"); // Working
});
```

**Comment by darrachequesne** [comment]
Hi! Did you check https://socket.io/docs/v3/troubleshooting-connection-issues/#In-the-browser?

**Reply by 0xBeycan** [reply]
Hello, the problem was that the connection was https, I did the things that should be done for https instead of http, but still failed.

**Comment by yakshdeveloper** [comment]
Question
Does Node.js and Ruby on Plesk support WebSockets (socket.io)?

How to provide WebSockets feature to customers?

Answer
Plesk itself does not forbid to use WebSockets, it only provides runtime environment to run Node.js and Ruby using Phusion Passenger, or iisnode.
The rest depends on particular website code design/implementation details.

On Linux
Note: Websockets work properly only if nginx-only web hosting is configured (with Proxy mode disabled) due to the following Apache module issue: [WebSockets don’t work on Apache](https://github.com/phusion/passenger/issues/1202).

[Log into Plesk](https://support.plesk.com/hc/en-us/articles/213413369)

Go to Websites & Domains > example.com > Apache & nginx Settings

In the nginx settings section, clear the Proxy mode checkbox

Note: In case this checkbox is not present, [install, or enable nginx](https://support.plesk.com/hc/en-us/articles/213944825).

Click on Apply

On Windows
[Log into Plesk](https://support.plesk.com/hc/en-us/articles/213413369)

Navigate to Domains > example.com > File Manager

Open the web.config file in the application’s document root, and add the following configuration to the <system.webServer> section:

<rewrite>
 <rules> 
 <rule name="SocketIO" patternSyntax="ECMAScript"> 
 <match url="socket.io.+" /> 
 <action type="Rewrite" url="app.js" /> 
 </rule> 
 </rules> 
 </rewrite>
Note: In case <rewrite> section already exists in the configuration file, instead add the <rule> part to the existing <rules>

The sample “Phusion Passenger: Node.js Web Socket demo” application can be used to verify whether Web Socket connections are working fine on the webserver: https://github.com/phusion/passenger-nodejs-websocket-demo


https://www.plesk.com/kb/support/does-node-js-on-plesk-support-websockets-socket-io/
